zap/src/main/java/org/parosproxy/paros/control/MenuFileControl.java

```diff
@@ -594,15 +598,35 @@ public String getDescription() {
                 if (file == null || !file.exists()) {
                     return;
                 }
-                // Import the context
-                Model.getSingleton().getSession().importContext(file);
-
-                // Show the dialog
-                View.getSingleton()
-                        .showSessionDialog(
-                                Model.getSingleton().getSession(),
-                                Constant.messages.getString("context.list"),
-                                true);
+                ZapXmlConfiguration config = new ZapXmlConfiguration(file);
+                String ctxName = config.getString(Context.CONTEXT_CONFIG_NAME);
+                Session session = Model.getSingleton().getSession();
+
+                while (session.getContext(ctxName) != null) {
+                    ctxName =
+                            (String)
+                                    JOptionPane.showInputDialog(
+                                            View.getSingleton().getMainFrame(),
+                                            Constant.messages.getString(
+                                                    "context.import.duplicate.dialog.message",
+                                                    ctxName),
+                                            Constant.messages.getString(
+                                                    "context.import.duplicate.dialog.title"),
+                                            JOptionPane.QUESTION_MESSAGE,
+                                            null,
+                                            null,
+                                            ctxName);
+                }
+                if (ctxName != null && !ctxName.isBlank()) {
```

Comment:
As discussed via IRC this should include handling for empty/null names on the off chance that a user edited or crafted the context file.